src/mlpack/methods/kernel_svm/kernel_svm.hpp

```diff
@@ -0,0 +1,209 @@
+   * @param numClass Number of classes for classification.
+   */
+  KernelSVM(const double regularization = 1.0,
+            const bool fitIntercept = false,
+            const double numClass = 2);
+
+  /**
+   * Classify the given points, returning the predicted labels for each point.
+   * The function calculates the probabilities for every class, given a data
+   * point. It then chooses the class which has the highest probability among
+   * all.
+   *
+   * @param data Set of points to classify.
+   * @param labels Predicted labels for each point.
+   */
+  void Classify(const MatType& data,
+                arma::Row<size_t>& labels) const;
+
+  /**
+   * Classify the given points, returning class scores and predicted
+   * class label for each point.
+   * The function calculates the scores for every class, given a data
+   * point. It then chooses the class which has the highest probability among
+   * all.
+   *
+   * @param data Matrix of data points to be classified.
+   * @param labels Predicted labels for each point.
+   * @param scores Class probabilities for each point.
+   */
+  void Classify(const MatType& data,
+                arma::Row<size_t>& labels,
+                arma::mat& scores) const;
+
+  /**
+   * Classify the given points, returning class scores for each point.
+   *
+   * @param data Matrix of data points to be classified.
+   * @param scores Class scores for each point.
+   */
+  void Classify(const MatType& data,
+                arma::mat& scores) const;
+
+  /**
+   * Classify the given point. The predicted class label is returned.
+   * The function calculates the scores for every class, given the point.
+   * It then chooses the class which has the highest probability among all.
+   *
+   * @param point Point to be classified.
+   * @return Predicted class label of the point.
+   */
+  template<typename VecType>
+  size_t Classify(const VecType& point) const;
+
+  /**
+   * Computes accuracy of the learned model given the feature data and the
+   * labels associated with each data point. Predictions are made using the
+   * provided data and are compared with the actual labels.
+   *
+   * @param testData Matrix of data points using which predictions are made.
+   * @param testLabels Vector of labels associated with the data.
+   * @return Accuracy of the model.
+   */
+  double ComputeAccuracy(const MatType& testData,
+                         const arma::Row<size_t>& testLabels) const;
+
+  /**
+   * Train the Kernel SVM with the given training data.
+   *
+   * @tparam OptimizerType Desired optimizer.
+   * @param data Input training features. Each column associate with one sample.
+   * @param labels Labels associated with the feature data.
+   * @param max_iter maximum number of iteration for training.
+   * @param tol tolerance value.
+   * @return Objective value of the final point.
+   */
+  double Train(const MatType& data,
+               const arma::Row<size_t>& labels,
```

Comment:
Here I am using `const arma::Row<size_t>& labels` this function will be directly accessed by user.